jck/jck-semiauto-updater/jckupdater.sh

```diff
@@ -167,50 +146,124 @@ isLatestUpdate() {
+			last_modified_dates+=("$artifactory_date")
+			file_last_modified["$artifactory_date"]=$file
+		done
+
+		artifactory_latest_modified=$(printf "%s\n" "${last_modified_dates[@]}" | sort -rV | head -1)
+		latest_file_name="${file_last_modified[$artifactory_latest_modified]}"
+		formatted_date=$(date -d "$artifactory_latest_modified" "+%Y-%m-%d") # reconvert month to value with leading zero for comparison
+		echo "Latest exclude file: $latest_file_name"
+		echo "Last Modified Date in GitHub: ${github_date_only}"
+		echo "Last Modified Date in Artifactory: ${formatted_date}"
+
+		# Compare and find the latest last modified date only if file is present in github. Else update exclude file
+		if [[ -n "${github_last_modified}" && "${github_date_only}" > "${formatted_date}" ]]; then
+			echo "No need to update exclude file in GitHub"
+			return 0
+		else
+			echo "Update exclude file with $latest_file_name from Artifactory"
+			return 1
+		fi
+	fi
+
+}
+
 ## Download directly from given URL under current folder
 getJCKSources() {
 	cd $WORKSPACE/jckmaterial
 	echo "remove build.txt file after comparison"
 	rm -rf build.txt
 	echo "download jck materials..."
 	
-	ARTIFACTORY_DOWNLOAD_URL=$ARTIFACTORY_DOWNLOAD_URL$JCK_UPDATE_NUMBER
-	echo $ARTIFACTORY_DOWNLOAD_URL
+	#download latest exclude file
+	if [ $exclude_files -eq 1 ]; then
+		curl -OLJSk -H X-JFrog-Art-Api:${ARTIFACTORY_TOKEN} ${ARTIFACTORY_JCK_DOWNLOAD_URL}exc/$latest_file_name && unzip -o "$latest_file_name"
+	fi
 
-	list #get list of files to download
-	
-	IFS=$'\n' read -r -d '' -a file_names_array <<< "$file_names"
+	if [ $result -ne 0 ]; then
+		ARTIFACTORY_JCK_UPDATE_NUMBER_URL=${ARTIFACTORY_JCK_DOWNLOAD_URL}tck/$JCK_UPDATE_NUMBER
+		echo $ARTIFACTORY_JCK_UPDATE_NUMBER_URL
 
-	if [ "${ARTIFACTORY_DOWNLOAD_URL}" != "" ]; then
-		for file in "${file_names_array[@]:1}"; do
-			url="$ARTIFACTORY_DOWNLOAD_URL$file"
-			executeCmdWithRetry "${file##*/}" "_ENCODE_FILE_NEW=UNTAGGED curl -OLJSk -H X-JFrog-Art-Api:${ARTIFACTORY_TOKEN} $url"
+		list "$ARTIFACTORY_JCK_UPDATE_NUMBER_URL" #get list of files to download
 
-			rt_code=$?
-			if [ $rt_code != 0 ]; then
-				echo "curl error code: $rt_code"
-				echo "Failed to retrieve $file. This is what we received of the file and MD5 sum:"
-				ls -ld $file
+		IFS=$'\n' read -r -d '' -a file_names_array <<< "$file_names"
 
-				exit 1
-			fi
-		done	
+		if [ "${ARTIFACTORY_JCK_UPDATE_NUMBER_URL}" != "" ]; then
+			for file in "${file_names_array[@]:1}"; do
+				url="$ARTIFACTORY_JCK_UPDATE_NUMBER_URL$file"
+				_ENCODE_FILE_NEW=UNTAGGED curl -OLJSk -H X-JFrog-Art-Api:${ARTIFACTORY_TOKEN} $url
+				rt_code=$?
+				if [ $rt_code != 0 ]; then
+					echo "curl error code: $rt_code"
+					echo "Failed to retrieve $file. This is what we received of the file and MD5 sum:"
+					ls -ld $file
+
+					exit 1
+				fi
+			done
+		fi
 	fi
 }
 
 #install Java
-get_JAVA_SDK(){
+getJavaSDK() {
```

Comment:
This is misleading. The function does get SDK or install java. It just tries to set JAVA_SDK_PATH=$WORKSPACE/../../../../jdkbinary/j2sdk-image if JAVA_HOME does not exist. And JAVA_HOME is a user-provided value. https://github.com/adoptium/aqa-tests/blob/a04569c44af4543f139c53c3aa123fe32ef750e1/jck/jck-semiauto-updater/jckupdater.sh#L52

Please remove this function completely. Instead of https://github.com/adoptium/aqa-tests/blob/a04569c44af4543f139c53c3aa123fe32ef750e1/jck/jck-semiauto-updater/jckupdater.sh#L52, user can provide JAVA_SDK_PATH. If not, the default is `$WORKSPACE/../../../../jdkbinary/j2sdk-image`.
Also remove L282 to 287, change it to `$JAVA_SDK_PATH/bin/java -jar $f -install shell_scripts -o $WORKSPACE/unpackjck`

Reply:
Changes done as per review. Removed `get_JAVA_SDK(). `
`$JAVA_SDK_PATH` is the default unless user provides value.